prefect_snowflake/credentials.py

```diff
@@ -119,3 +119,49 @@ def _validate_okta_kwargs(cls, values):
                 "`okta_endpoint` must be provided"
             )
         return values
+
+
+def resolve_pem_certificate(private_key: Union[str, bytes], password: Optional[str]):
```

Comment:
Perhaps this can be a private function?
```suggestion
def _resolve_pem_certificate(private_key: Union[str, bytes], password: Optional[str]):
```

Also, I was wondering what you think about moving this into a class method like `SnowflakeCredentials`? Then can use `root_validator` to perform this check.

Reply:
Will resolve in next commit